opentelemetry-sdk/src/opentelemetry/sdk/_metrics/view.py

```diff
@@ -0,0 +1,111 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=protected-access
+
+from logging import getLogger
+from re import match
+
+_logger = getLogger(__name__)
+
+
+class View:
+    def __init__(
+        self,
+        instrument_type=None,
+        instrument_name=None,
+        meter_name=None,
+        meter_version=None,
+        meter_schema_url=None,
+        name=None,
+        description=None,
+        attribute_keys=None,
+        extra_dimensions=None,
+        aggregation=None,
+        exemplar_reservoir=None,
+    ):
+        if (
+            name
+            is instrument_type
+            is instrument_name
+            is meter_name
+            is meter_version
+            is meter_schema_url
+            is None
+        ):
+            raise Exception(
+                "Some instrument selection criteria must be provided"
+            )
+
+        self._instrument_type = instrument_type
+        self._instrument_name = instrument_name
+        self._meter_name = meter_name
+        self._meter_version = meter_version
+        self._meter_schema_url = meter_schema_url
+
+        self._name = name
+
+        if self._name is not None:
+            if self._instrument_type is not None:
+                _logger.warning(
+                    "Instrument type matching disabled "
+                    "for view with a defined name"
+                )
+                self._instrument_type = None
+
+            if self._instrument_name is not None:
```

Comment:
The wording of the spec was tweaked in this PR from yesterday https://github.com/open-telemetry/opentelemetry-specification/pull/2306

I think there are more cases here that need to be covered for "selectors that select more than one instrument".

Is it specified how we should handle this error case? It looks like this PR tries to recover, but we could also ignore the view altogether or raise an exception during instantiation. Wdyt?

Reply:
Ok, I have added restrictions to more criteria. Why do you think the PR tries to recover? If so, how?

Reply:
I may be misreading the code, but it look like it takes the first match in these cases instead of raising an error for example

Reply:
Sorry, don't understand, can you please give more details?

Reply:
Ok, made them raise an error instead.